lib/components/AlbumScreen/song_menu.dart

```diff
@@ -266,90 +339,129 @@ class _SongMenuState extends State<SongMenu> {
+                                    : AppLocalizations.of(context)!
+                                        .sleepTimerTooltip,
+                                iconColor: timerValue != null
+                                    ? iconColor
+                                    : Theme.of(context)
+                                            .textTheme
+                                            .bodyMedium
+                                            ?.color ??
+                                        Colors.white,
+                              );
+                            },
+                          ),
+                          // [Playback speed widget will be added here if conditions are met]
+                          PlaybackAction(
+                            icon: loopModeIcons[playbackBehavior.loop]!,
+                            onPressed: () async {
+                              _queueService.toggleLoopMode();
+                            },
+                            tooltip: loopModeTooltips[playbackBehavior.loop]!,
+                            iconColor:
+                                playbackBehavior.loop == FinampLoopMode.none
+                                    ? Theme.of(context)
+                                            .textTheme
+                                            .bodyMedium
+                                            ?.color ??
+                                        Colors.white
+                                    : iconColor,
+                          ),
+                        ];
+
+                        final speedWidget = PlaybackAction(
+                          icon: TablerIcons.brand_speedtest,
+                          onPressed: () {
+                            toggleSpeedMenu();
+                          },
+                          tooltip: playbackSpeedTooltip,
+                          iconColor: playbackBehavior.speed == 1.0
+                              ? Theme.of(context).textTheme.bodyMedium?.color ??
+                                  Colors.white
+                              : iconColor,
                         );
+
+                        if (speedWidgetWasVisible) {
+                          sliverArray.insertAll(2, [speedWidget]);
+                          return SliverCrossAxisGroup(
+                            slivers: sliverArray,
+                          );
+                        }
+                        return FutureBuilder<bool>(
+                            future:
+                                shouldShowSpeedWidget(playbackBehavior.speed),
+                            builder: (context, snapshot) {
+                              if (snapshot.connectionState ==
+                                      ConnectionState.done &&
+                                  snapshot.data == true) {
+                                speedWidgetWasVisible = true;
+                                sliverArray.insertAll(2, [speedWidget]);
```

Comment:
I'm not a big fan of this, it makes it hard to read the code and find the relevant section for the widget. Can you try supplying the sliverArray directly to the `SliverCrossAxisGroup`, and using the FutureBuilder there, so that it's in the right position? You can return a `SliverPadding` as long as the speed controls shouldn't be shown.

Reply:
I think the reason for me doing it this way was that simply adding the `FutureBuilder` to the `CrossAxisGroup` already takes up the typical button space (like a child `<div>` with `align-items: stretch` ;)). Maybe there is a way to prevent that, I just haven't found it yet.

Reply:
I see. I guess it's fine for now...